### PR TITLE
Update In-tree storage driver status

### PR DESCRIPTION
**Pull request by mengjiao-liu**
Follow up on some comments from  PR #42302 

/sig storage

/cc @xing-yang 
for technical review

### CODE REVIEW

**Comment by tengqm** [comment]
/approve
A tech review is desirable. @xing-yang

**Comment by mengjiao-liu** [comment]
ping @xing-yang

**Comment by divya-mohan0209** [comment]
@kubernetes/sig-storage-pr-reviews : PTAL!
Alternately, @xing-yang whenever you have time, please could you review this?

**Comment by xing-yang** [comment]
> https://github.com/orgs/kubernetes/teams/sig-storage-pr-reviews : PTAL!
Alternately, @xing-yang whenever you have time, please could you review this?

@divya-mohan0209 I'll take a look. Thanks!

File: content/en/docs/concepts/storage/persistent-volumes.md [hunk 1]
```diff
@@ -185,7 +185,7 @@ another claim because the previous claimant's data remains on the volume.
 An administrator can manually reclaim the volume with the following steps.
 
 1. Delete the PersistentVolume. The associated storage asset in external infrastructure
-   (such as an AWS EBS or GCE PD volume) still exists after the PV is deleted.
+   (such as a GCE PD volume) still exists after the PV is deleted.
```
Comment (xing-yang) [inline content/en/docs/concepts/storage/persistent-volumes.md:188]: GCE PD in-tree volume plugin was removed in 1.28. Do we still need this in the doc on the main branch?

Comment (mengjiao-liu) [reply content/en/docs/concepts/storage/persistent-volumes.md]: Removed GCE PD doc in this file.

**Comment by mengjiao-liu** [comment]
/label tide/merge-method-squash

Preview page:
- https://deploy-preview-42415--kubernetes-io-main-staging.netlify.app/docs/concepts/storage/volumes/#gcepersistentdisk
- https://deploy-preview-42415--kubernetes-io-main-staging.netlify.app/docs/concepts/storage/persistent-volumes/

File: content/en/docs/concepts/storage/storage-classes.md
Comment (mengjiao-liu) [inline content/en/docs/concepts/storage/storage-classes.md:1]: https://kubernetes.io/docs/concepts/storage/storage-classes/#allowed-topologies
```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: standard
provisioner: kubernetes.io/gce-pd
parameters:
  type: pd-standard
volumeBindingMode: WaitForFirstConsumer
allowedTopologies:
- matchLabelExpressions:
  - key: topology.kubernetes.io/zone
    values:
    - us-central-1a
    - us-central-1b
```

The example contains `kubernetes.io/gce-pd`, but the  GCEPersistentDisk in-tree volume has been removed, which may confuse users as an example.

> The following plugins support WaitForFirstConsumer with dynamic provisioning:
GCEPersistentDisk
The following plugins support WaitForFirstConsumer with pre-created PersistentVolume binding:
All of the above
Local
CSI volumes are also supported with dynamic provisioning and pre-created PVs, but you'll need to look at the documentation for a specific CSI driver to see its supported topology keys and examples.

From[ the Volume Binding Mode section](https://kubernetes.io/docs/concepts/storage/storage-classes/#volume-binding-mode), we know that `WaitForFirstConsumer` is now only supported by `Local` and `CSI volumes`, 
perhaps sig-storage can provide an example use `Local` or other?  I am not sure how to change it. @xing-yang

**Pull request merged**